qiskit_experiments/framework/base_experiment.py

```diff
@@ -167,29 +144,114 @@ def _initialize_experiment_data(
 
         return experiment_data._copy_metadata()
 
-    def run_analysis(self, experiment_data, **options) -> ExperimentData:
+    def _pre_transpile_hook(self, backend: Backend):
+        """An extra subroutine executed before transpilation.
+
+        Args:
+            backend: Target backend.
+        """
+        pass
+
+    # pylint: disable = unused-argument
+    def _post_transpile_hook(
+        self, circuits: List[QuantumCircuit], backend: Backend
+    ) -> List[QuantumCircuit]:
+        """An extra subroutine executed after transpilation.
+
+        Args:
+            circuits: List of transpiled circuits.
+            backend: Target backend.
+
+        Returns:
+            List of circuits to execute.
+        """
+        return circuits
+
+    def run_transpile(self, backend: Backend, **options) -> List[QuantumCircuit]:
+        """Run transpile and returns transpiled circuits.
+
+        Args:
+            backend: Target backend.
+            options: User provided runtime options.
+
+        Returns:
+            Transpiled circuit to execute.
+        """
+        # Run pre transpile. This is implemented by each experiment subclass.
+        self._pre_transpile_hook(backend)
+
+        # Get transpile options
+        transpile_options = copy.copy(self.transpile_options)
+        transpile_options.update_options(
+            initial_layout=list(self._physical_qubits),
+            **options,
+        )
+        transpile_options = transpile_options.__dict__
+
+        circuits = transpile(circuits=self.circuits(backend), backend=backend, **transpile_options)
+
+        # Run post transpile. This is implemented by each experiment subclass.
+        circuits = self._post_transpile_hook(circuits, backend)
+
+        return circuits
+
+    def _post_analysis_hook(self, experiment_data: ExperimentData):
```

Comment:
So we could use this #251 to perform calibration updating?

Reply:
Yes. Perhaps we can use post/pre transpile for generating pulse gate (pre: update instmap / post: directly attach pulse gate).

Reply:
Yes, I think that is a good idea.